vector-im/element-android
# Mutualize the pending auth handling (PSG-742)

## Type of change


## Content

Creation of a new class `PendingAuthHandler` to mutualize the handling of pending auth for actions which require authentication. It was previously used in several `ViewModel`. I am doing this refactor to be able to test this and because we will need it for sign out action in the new device management screens.

## Motivation and context


Closes #7193 

## Screenshots / GIFs





## Tests



- Go to a screen using re-auth for example the device manager screen
- Go to settings -> Security & Privacy -> Show all sessions
- Long press on of your devices
- Press the sign out action
- Check the authentication flow is working well

## Tested devices

- OS version(s): Android 11

## Checklist
